Add submitter attribute to ArchiveItems

oneroyalace opened the pull request:
This PR adds adds a new relationship to the data model: `ArchiveItem belongs_to user`. 

When an `ArchiveItem` is added, we'll record the user who submitted it. If that user's account is deleted, all the `ArchiveItem`s they submitted will have their `submitter_id` attr nullified: 

https://github.com/TechAndCheck/zenodotus/blob/fdb94484f46276e9feb20f024f0c560bd52e2bc5/app/models/user.rb#L3

I'm submitting this PR in service of #112, which adds a feature allowing users to delete their accounts

# Testing
1. `rails db:migrate`
2. `rails t`

cguess commented:
I'm getting 12 errors when this test runs:

```
ActiveRecord::AssociationTypeMismatch: User(#22820) expected, got #<Sources::InstagramUser id: "9a35c656-4d6b-46f4-bd12-c15bdf995b67", display_name: "snoopdogg", handle: "snoopdogg", number_of_posts: 51755, followers_count: 65661962, following_count: 5327, verified: true, profile: "ALGORITHM 11/19/21", url: "https://snoopdogg.lnk.to/AlgorithmIB", profile_image_url: "https://scontent-lga3-1.cdninstagram.com/v/t51.288...", profile_image_data: {"id"=>"b2d510f882f4c4e653e7d2a2ba4b950d", "storage"=>"store", "metadata"=>{"filename"=>"20211217-4288-depq31", "size"=>5618, "mime_type"=>nil}}, created_at: "2021-12-17 11:28:32.768518000 +0000", updated_at: "2021-12-17 11:28:32.772866000 +0000"> which is an instance of Sources::InstagramUser(#25280)
```

Can you take a look before we merge? Zorki is running correctly so it's something to do with the code base

oneroyalace commented:
> I'm getting 12 errors when this test runs:
> 
> ```
> ActiveRecord::AssociationTypeMismatch: User(#22820) expected, got #<Sources::InstagramUser id: "9a35c656-4d6b-46f4-bd12-c15bdf995b67", display_name: "snoopdogg", handle: "snoopdogg", number_of_posts: 51755, followers_count: 65661962, following_count: 5327, verified: true, profile: "ALGORITHM 11/19/21", url: "https://snoopdogg.lnk.to/AlgorithmIB", profile_image_url: "https://scontent-lga3-1.cdninstagram.com/v/t51.288...", profile_image_data: {"id"=>"b2d510f882f4c4e653e7d2a2ba4b950d", "storage"=>"store", "metadata"=>{"filename"=>"20211217-4288-depq31", "size"=>5618, "mime_type"=>nil}}, created_at: "2021-12-17 11:28:32.768518000 +0000", updated_at: "2021-12-17 11:28:32.772866000 +0000"> which is an instance of Sources::InstagramUser(#25280)
> ```
> 
> Can you take a look before we merge? Zorki is running correctly so it's something to do with the code base

Fixed. I added an argument to `create_from_zorki_hash` named `user` without realizing that the same variable name was already being used in the function. 
https://github.com/TechAndCheck/zenodotus/blob/903200401297123fbbd3c4b88f53556f17ecbcf9/app/models/sources/instagram_post.rb#L80-L83

cguess commented on app/views/archive/index.html.erb:
```diff
@@ -9,11 +9,11 @@
-          ↓
-        <% end %>
-      </a>
+<!--      <a href="#">-->
```
Should we just remove this instead of commenting it out?

oneroyalace replied:
Removed

cguess reviewed (approved):
One little thing, but otherwise, consider this approved.
